components/script/script_thread.rs

```diff
@@ -218,6 +219,26 @@ impl Drop for ScriptUserInteractingGuard {
     }
 }
 
+#[derive(Clone)]
+/// A weak handle to the current script thread.
+/// Get an Rc to it with the upgrade method.
+/// Use #[custom_tracer] in your structure.
+pub struct WeakScriptHandle(Weak<ScriptThread>);
```

Comment:
Wouldn't here make more sense to simply impl JSTraceable directly, so user of this does not need to do anything special (also currently nobody actually uses custom_trace).